idf_build_apps/build_job.py

```diff
@@ -12,9 +15,18 @@
 )
 
 
-class BuildJob(BaseModel):
+class OverridePriority(Enum):
+    OVERRIDE_SDKCONFIG = 'override_sdkconfig'
+    OVERRIDE_SDKCONFIG_FILE = 'override_sdkconfig_file'
```

Comment:
I think we could hard code it. `OVERRIDE_SDKCONFIG_ITEM` should override the settings set in `OVERRIDE_SDKCONFIG_FILE`

otherwise you have to provide a cli args to make full use of it, but for users, it would be very hard to understand the logic behind that.

we may get questions like, if we choose `--override-priority file_first`, will the `App().override_sdkconfig_items` override the `BuildAppJob.override_sdkconfig_file_names`?

We could simplify it a bit to

BuildAppJob.override_sdkconfig_items could be calculated by

- items in `--override-sdkconfig-filenames`
- (higher priority) items in `--override-sdkconfig-items`

Property `App().override_sdkconfig_items` could be calculated by

- items in `BuildAppJob.override_sdkconfig_items`
- (higher priority) items in `App()._override_sdkconfig_items`

Reply:
The OverridePriority has been removed. To determine priority `(higher priority) items in --override-sdkconfig-items` was used.